src/zkevm_specs/evm/execution/return_revert.py

```diff
@@ -26,14 +28,21 @@ def return_(instruction: Instruction):
 
     rwc_delta = 3
 
-    if instruction.curr.is_create:
+    if instruction.curr.is_create and is_success:
         # A. Returns the specified memory chunk as deployment code.
 
         # TODO: Untested case.  Test it once create Tx is implemented, and once
         # CREATE/CREATE2 are implemented.
+        callee_address = instruction.call_context_lookup(CallContextFieldTag.CalleeAddress)
+        reversion_info = instruction.reversion_info()
+        code_hash, code_hash_prev = instruction.account_write(
```

Comment:
I have some questions about this part.
When an contract account is created several things happen:
1. A code_hash is calculated from the returned byte array
2. The code_hash is set to the account
3. The nonce is set to 1

Here you're doing (2).  What about (1) and (3)?

Reply:
Thanks for catching this up @ed255 

I thought this was just a renaming PR but seems that it has much more on it :confused:

Reply:
(1) code_hash must be the keccak of the bytecode because of the copy table lookup. This in turn forces the (code_hash, bytecode) to be in the bytecode table, where the keccak lookup happens.
(3) (will) happen in the create/create2/begin_tx step, at the same time that the `value` is transferred to the new address. In the bus mapping for CREATE this already is implemented at https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/83bfbef71fc006fab4a07fc8a830745e916055bc/bus-mapping/src/evm/opcodes/create.rs#L102

Reply:
> (1) code_hash must be the keccak of the bytecode because of the copy table lookup. This in turn forces the (code_hash, bytecode) to be in the bytecode table, where the keccak lookup happens. 

Oh right! I missed the connection of the `code_hash` with the copy table lookup, thanks for the clarification!

> (3) (will) happen in the create/create2/begin_tx step, at the same time that the `value` is transferred to the new address. In the bus mapping for CREATE this already is implemented at https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/83bfbef71fc006fab4a07fc8a830745e916055bc/bus-mapping/src/evm/opcodes/create.rs#L102

Thanks a lot for the explanation!